kv/kvserver: TestMultiSSTWriterInitSST failed

kv/kvserver.TestMultiSSTWriterInitSST [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19397760?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19397760?buildTab=artifacts#/) on master @ [8cd345b3f2966e8a012641abbe72171f19ad1a9b](https://github.com/cockroachdb/cockroach/commits/8cd345b3f2966e8a012641abbe72171f19ad1a9b):


```
=== RUN   TestMultiSSTWriterInitSST
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/15bcc1d4cfe2c01885bd956ef4f14c71/logTestMultiSSTWriterInitSST1579220335
    test_log_scope.go:76: use -show-logs to present logs inline
    replica_sst_snapshot_storage_test.go:311: 
          /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/31967/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/testdata/echotest/TestMultiSSTWriterInitSST:1:
          echo
        output didn't match expected:
          @@ -1,6 +1,6 @@
          -writeBytes=4740 sstSize=4872 dataSize=132
          +writeBytes=4392 sstSize=4524 dataSize=132
           >> sst0:
           rangedel: /Local/RangeID/0/{r""-s""}
           rangekeydel: /Local/RangeID/0/{r""-s""}
           >> sst1:
           rangedel: /Local/Range{"d"-/Max}
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/15bcc1d4cfe2c01885bd956ef4f14c71/logTestMultiSSTWriterInitSST1579220335
--- FAIL: TestMultiSSTWriterInitSST (0.09s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiSSTWriterInitSST.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-49321

kv/kvserver.TestMultiSSTWriterInitSST [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19399186?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19399186?buildTab=artifacts#/) on master @ [30bcd20474b55d698bff34fc5dd8bb963c7f8fcc](https://github.com/cockroachdb/cockroach/commits/30bcd20474b55d698bff34fc5dd8bb963c7f8fcc):


```
=== RUN   TestMultiSSTWriterInitSST
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/7334302ae8c22802de8334e77d6901c9/logTestMultiSSTWriterInitSST1026962291
    test_log_scope.go:76: use -show-logs to present logs inline
    replica_sst_snapshot_storage_test.go:311: 
          /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/30024/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/testdata/echotest/TestMultiSSTWriterInitSST:1:
          echo
        output didn't match expected:
          @@ -1,6 +1,6 @@
          -writeBytes=4740 sstSize=4872 dataSize=132
          +writeBytes=4392 sstSize=4524 dataSize=132
           >> sst0:
           rangedel: /Local/RangeID/0/{r""-s""}
           rangekeydel: /Local/RangeID/0/{r""-s""}
           >> sst1:
           rangedel: /Local/Range{"d"-/Max}
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/7334302ae8c22802de8334e77d6901c9/logTestMultiSSTWriterInitSST1026962291
--- FAIL: TestMultiSSTWriterInitSST (0.06s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiSSTWriterInitSST.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/kv/kvserver/replica_sst_snapshot_storage_test.go
+++ b/pkg/kv/kvserver/replica_sst_snapshot_storage_test.go
@@ -284,8 +284,13 @@ func TestMultiSSTWriterInitSST(t *testing.T) {
 	localSpans := keySpans[:len(keySpans)-1]
 	mvccSpan := keySpans[len(keySpans)-1]
 
+	st := cluster.MakeTestingClusterSettings()
+
+	// Disabling columnar blocks causes stats changes.
+	storage.ColumnarBlocksEnabled.Override(context.Background(), &st.SV, true)
+
 	msstw, err := newMultiSSTWriter(
-		ctx, cluster.MakeTestingClusterSettings(), scratch, localSpans, mvccSpan, 0,
+		ctx, st, scratch, localSpans, mvccSpan, 0,
 		false /* skipRangeDelForMVCCSpan */, false, /* rangeKeysInOrder */
 	)
 	require.NoError(t, err)

